java/com/google/javascript/jscomp/JsCompilerRunner.java

```diff
@@ -65,6 +68,81 @@ protected CompilerOptions createOptions() {
         options.setPropertyRenaming(PropertyRenamingPolicy.OFF);
         options.setDisambiguateProperties(false);
     }
+    if (devBuild) {
+      applySpeedyOptions(options);
+    }
     return options;
   }
+
+  /**
+   * Remove checks and undo most of the options set by the ADVANCED
+   * CompilationLevel to make development builds faster.
+   *
+   * Does not change property renaming options so code that works when compiled
+   * this way will also work with property renaming.
+   *
+   * @param options CompilerOptions object to update
+   */
+  private void applySpeedyOptions(CompilerOptions options) {
+    options.setLanguageOut(CompilerOptions.LanguageMode.ECMASCRIPT_2020);
+
+    options.resetWarningsGuard();
+
+    options.setCheckSymbols(false);
+    options.setCheckTypes(false);
```

Comment:
Type-check errors seem like they come up often enough that it might be worth leaving this on, if it doesn't tank the timings

Reply:
This is one of the most time-consuming parts of a build and turning it on adds ~50% more time spent in Closure.